include/audit_events.hpp

```diff
@@ -38,32 +41,59 @@ inline void auditEvent([[maybe_unused]] const crow::Request& req,
         return;
     }
 
-    strcpy(cnfgBuff, opPath);
+    opPathLen = std::strlen(opPath) + 1;
+    if (opPathLen > bufLeft)
+    {
+        // Truncate event message to fit into fixed sized buffer.
+        BMCWEB_LOG_WARNING << "Audit buffer too small, truncating:"
+                           << " bufLeft=" << bufLeft
+                           << " opPathLen=" << opPathLen;
+        opPathLen = bufLeft;
+        code = __LINE__;
+    }
+    strncpy(cnfgBuff, opPath, opPathLen);
+    cnfgBuff[opPathLen - 1] = '\0';
+    bufLeft -= opPathLen;
 
     // encode user account name to ensure it is in an appropriate format
-    user = audit_encode_nv_string("acct", req.session->username.c_str(), 0);
+    user = audit_encode_nv_string("acct", userName.c_str(), 0);
     if (user == NULL)
     {
         BMCWEB_LOG_ERROR << "Error appending user to audit msg : "
                          << strerror(errno);
+        code = __LINE__;
     }
     else
     {
-        strncat(cnfgBuff, user, 50);
+        userLen = std::strlen(user) + 1;
+
+        if (userLen > bufLeft)
+        {
+            // Username won't fit into fixed sized buffer. Leave it off.
+            BMCWEB_LOG_WARNING << "Audit buffer too small, truncating:"
```

Comment:
Should this log message be updated? The comment says to "Leave it off" but the log is printing ", truncating", although there's no truncate logic being done.

Reply:
Good point. I pushed another commit which rewords the warning message.